Plugins/NativeCamera/Source/NativeCamera.cpp

```diff
@@ -0,0 +1,117 @@
+#include <napi/napi.h>
+#include "NativeCamera.h"
+#include "NativeVideo.h"
+
+#include <Babylon/JsRuntime.h>
+#include <GraphicsImpl.h>
+#include <vector>
+#include <algorithm>
+
+namespace Babylon::Plugins::Internal
+{
+    class NativeCamera : public Napi::ObjectWrap<NativeCamera>
+    {
+        static constexpr auto JS_NAVIGATOR_NAME = "navigator";
+        static constexpr auto JS_CLASS_NAME = "_NativeCamera";
+        static constexpr auto JS_NATIVECAMERA_CONSTRUCTOR_NAME = "NativeCamera";
+
+    public:
+
+        static void Initialize(Napi::Env env)
+        {
+            Napi::HandleScope scope{ env };
+
+            Napi::Function func = NativeCamera::DefineClass(
+                env,
+                JS_CLASS_NAME,
+                {
+                    NativeCamera::InstanceMethod("createVideo", &NativeCamera::CreateVideo),
+                    NativeCamera::InstanceMethod("updateVideoTexture", &NativeCamera::UpdateVideoTexture),
+                });
+
+            JsRuntime::NativeObject::GetFromJavaScript(env).Set(JS_NATIVECAMERA_CONSTRUCTOR_NAME, func);
+
+            // create or get global navigator object
+            Napi::Object global = env.Global();
+            Napi::Object navigator;
+            if (global.Has(JS_NAVIGATOR_NAME))
+            {
+                navigator = global.Get(JS_NAVIGATOR_NAME).As<Napi::Object>();
+            }
+            else
+            {
+                navigator = Napi::Object::New(env);
+                global.Set(JS_NAVIGATOR_NAME, navigator);
+            }
+
+            // append media devices to navigator
+            Napi::Object mediaDevices = Napi::Object::New(env);
```

Comment:
nit: Should we check to make sure this doesn't exist before creating a new one? Probably doesn't matter right now, just thinking this might get confusing if something else ever tries to make media devices too.